### PR TITLE
Add support for the AWS S3 gstreamer plugin

### PR DESCRIPTION
**Pull request by brgl**
*Description of changes:*

Add a recipe for the S3 gstreamer plugin and another one to satisfy its dependency on the C++ SDK.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by brgl** [comment]
I don't have access to the CI output, don't know what's wrong.

**Comment by rpcme** [comment]
Hello @brgl it seems like some PR checks have failed. @nateglims can we provide some details?

Also note that the AWS SDK for C++ was in plan but actually the compilation options for specifying individual modules, likely through PACKAGECONFIG, should be provided. I am ok with this as an initial contribution but I will need to spend some time defining sane defaults while allowing users to tune the build to specific modules - for example, if a user only wants to build and distribute SDK capabilities for a specific subset of services.

**Comment by nateglims** [comment]
Checklayer had the following error:

```
ERROR: ParseError in configuration INHERITs: Could not inherit file classes/image-prelink.bbclass
```

**Comment by nateglims** [comment]
The issue seems to be upstream, I am investigating.

**Comment by brgl** [comment]
> Hello @brgl it seems like some PR checks have failed. @nateglims can we provide some details?
> 
> Also note that the AWS SDK for C++ was in plan but actually the compilation options for specifying individual modules, likely through PACKAGECONFIG, should be provided. I am ok with this as an initial contribution but I will need to spend some time defining sane defaults while allowing users to tune the build to specific modules - for example, if a user only wants to build and distribute SDK capabilities for a specific subset of services.

I considered `PACKAGECONFIG` options for every module but the way I see it: `PACKAGECONFIG` is much more awkward to use than providing additional packages in the recipe and since aws-sdk-cpp doesn't seem to have a lot of external dependencies, it may be fine to just build everything, put it into separate packages and let the user pick and choose.

File: recipes-sdk/aws-sdk-cpp/aws-sdk-cpp_1.9.185.bb [hunk 1]
```diff
@@ -0,0 +1,39 @@
+# -*- mode: Conf; -*-
+SUMMARY = "AWS C++ SDK"
+HOMEPAGE = "https://github.com/aws/aws-sdk-cpp"
+LICENSE = "Apache-2.0"
+LIC_FILES_CHKSUM = "file://LICENSE;md5=e3fc50a88d0a364313df4b21ef20c29e"
+
+SRC_URI = " \
+    gitsm://github.com/aws/aws-sdk-cpp.git;protocol=https;branch=main;tag=1.9.185 \
```
Comment (nateglims) [inline recipes-sdk/aws-sdk-cpp/aws-sdk-cpp_1.9.185.bb:8]: Though most amazon SDK's use tags, we are switching our yocto recipes to strictly use `SRC_REV`. Can we switch to that here? I looks like the rev is `9c7d9735e4fd9f018bfab79ef0d5890bbf2b58f3` for this release.

**Comment by rpcme** [comment]
> I considered `PACKAGECONFIG` options for every module but the way I see it: `PACKAGECONFIG` is much more awkward to use than providing additional packages in the recipe and since aws-sdk-cpp doesn't seem to have a lot of external dependencies, it may be fine to just build everything, put it into separate packages and let the user pick and choose.

This is an interesting option

**Comment by brgl** [comment]
The new version I just pushed has several improvements:

- dropped the tag from `SRC_URI` in favor of the git commit hash in `SRCREV`
- dropped `gitsm://` from `SRC_URI` as we don't need to fetch the submodules - we get them from other packages in `meta-aws`
- added a third patch that extends the `aws-sdk-cpp` recipe with some popular modules and also serves to showcase how bitbake handles the library dependencies implicitly if the packages come from a single recipe, e.g. the `access-management` module implicitly pulls in `iam`

File: recipes-sdk/aws-sdk-cpp/aws-sdk-cpp_1.9.185.bb (hunk 1 above)
Comment (nateglims) [reply recipes-sdk/aws-sdk-cpp/aws-sdk-cpp_1.9.185.bb]: Thanks!

**Review by nateglims — APPROVED** [review]

**Pull request merged**